### ISSUE TITLE
[Bug]: uwsgi module not working with Unix sockets

### ISSUE THREAD

**Issue by bartekre** [netdata/netdata#18617]
### Bug description

When configured to fetch uWSGI stats via UDS (Unix Domain Socket), the collector fails with a broken pipe error:
```
failed to query stats: write unix @->/run/uwsgi/stats/stats.sock: write: broken pipe
```

This issue appears to be caused by an unnecessary write operation: https://github.com/netdata/netdata/blob/99eed4690a35890a746b4f986e4d7de95a5c1d97/src/go/plugin/go.d/modules/uwsgi/client.go#L45

uWSGI does not seem to read from the stats socket at all, which leads to a broken pipe when the collector attempts to write. Notably, this issue doesn't occur when using TCP sockets. In that case, it seems that the bytes written by the collector are simply ignored by uWSGI, allowing the connection to work without errors.

### Expected behavior

The uWSGI collector should fetch stats by reading from the socket without attempting to write to it.

### Steps to reproduce

1. Configure uWSGI to serve stats via UDS by adding the following line to `/etc/uwsgi/apps-enabled/myapp.ini`:
```
stats = /run/uwsgi/stats.sock
```
2. Configure the uWSGI module in Netdata (`/etc/netdata/go.d/uwsgi.conf`) to connect to the UDS socket:
```
jobs:
  - name: "myapp"
    address: "unix:///run/uwsgi/stats.sock"
```
3. Restart the Netdata service
```
systemctl restart netdata.service
```
4. Check the error logs and observe the broken pipe error:
```
# journalctl _SYSTEMD_INVOCATION_ID="$(systemctl show --value --property=InvocationID netdata)" --namespace=netdata --grep uwsgi
Sep 25 13:26:33 myapp netdata[2605004]: level=error msg="failed to query stats: write unix @->/run/uwsgi/stats.sock: write: broken pipe" plugin=go.d c>
Sep 25 13:26:33 myapp netdata[2605004]: level=error msg="check failed" plugin=go.d collector=uwsgi job=myapp
Sep 25 13:26:40 myapp netdata[2605004]: level=info msg="check success" plugin=go.d collector=uwsgi job=local
Sep 25 13:26:40 myapp netdata[2605004]: level=info msg="started, data collection interval 1s" plugin=go.d collector=uwsgi job=local
Sep 25 13:26:42 myapp netdata[2604787]: ALERT 'apps_group_file_descriptors_utilization' of instance 'app.uwsgi_fds_open_limit' on node 'myapp', transit>


### Installation method

kickstart.sh

### System info

```shell
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=20.04
/etc/lsb-release:DISTRIB_CODENAME=focal
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 20.04.6 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="20.04.6 LTS (Focal Fossa)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 20.04.6 LTS"
/etc/os-release:VERSION_ID="20.04"
/etc/os-release:VERSION_CODENAME=focal
/etc/os-release:UBUNTU_CODENAME=focal
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.47.2
    Installation Type __________________________________________ : binpkg-deb
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ : ubuntu 20.04
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 5.4.0-167-generic
    Operating System ___________________________________________ : Ubuntu
    Operating System ID ________________________________________ : ubuntu
    Operating System ID Like ___________________________________ : debian
    Operating System Version ___________________________________ : 20.04.6 LTS (Focal Fossa)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 1
    CPU Frequency ______________________________________________ : 2095000000
    RAM Bytes __________________________________________________ : 8315625472
    Disk Capacity ______________________________________________ : 53687091200
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : microsoft
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
    libmnl (library for working with netfilter) ________________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : YES
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : YES
    Xen VBD Error Tracking _____________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : YES
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

#### Possible workarounds

1. Use TCP socket instead
2. Redirect writes to `/dev/null` with socat:
```
socat UNIX-LISTEN:/run/uwsgi/stats-ro.sock,fork 'UNIX-CONNECT:/run/uwsgi/stats.sock!!PIPE:/dev/null'
```

**Comment by ilyam8** [comment]
Hi, @bartekre.

> This issue appears to be caused by an unnecessary write operation:

This line doesn't write to the socket but writes response data to the local buffer (read response).

**Comment by bartekre** [comment]
> Hi, @bartekre.
> 
> > This issue appears to be caused by an unnecessary write operation:
> 
> This line doesn't write to the socket but writes response data to the local buffer (read response).

Ah, sorry `c.conn.Command()` call one line above does a write. I fixed the permalink.
Even though the output string is empty, the write still seems to be performed. I was able to reproduce this error in Python:
```
>>> import socket
>>> sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
>>> sock.connect("/run/uwsgi/stats.sock")
>>> sock.send(b"")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
BrokenPipeError: [Errno 32] Broken pipe
```
I'm still able to call `recv()` afterwards, so just ignoring the error seems to do the job

**Comment by ilyam8** [comment]
I can't reproduce it with my instance but `connect; read` instead of `connect; write; read` worked for me too. Fixed in #18618.

**Comment by bartekre** [comment]
That was fast, thank you!
